apollo-federation/src/sources/connect/json_selection/apply_to.rs

```diff
@@ -1238,12 +1301,18 @@ mod tests {
                 }))],
             ),
         );
+
+        // A single variable path should not be mapped over an input array.
+        assert_eq!(
+            selection!("$args.id").apply_with_vars(&json!([1, 2, 3]), &vars),
+            (Some(json!("id from args")), vec![]),
+        );
     }
 
     #[test]
     fn test_apply_to_variable_expressions_typename() {
         let typename_object =
-            selection!("__typename: $typenames.Product reviews { __typename: $typenames.Review }")
+            selection!("__typename: $->echo('Product') reviews { __typename: $->echo('Review') }")
```

Comment:
Oh yup, echo! I think that's a cleaner solution, but we either need to hide these in the debug info or be okay exposing this syntax before it's ready for users. Hopefully that window is small, and we can prioritize validations early next week?

Reply:
I guess I would vote to hide in debug info, since I'm not clear on the value of exposing that info.